util/ivmFactory.js

```diff
@@ -38,31 +38,64 @@ async function createIvm(code, libraryVersionIds) {
       const eventMessages = events.map(event => event.message);
       const eventsMetadata = {};
       events.forEach(ev => {
-        eventsMetadata[ev.message.messageId] = ev.metadata;
+        eventsMetadata[ev.metadata ? ev.metadata.messageId : ev.message.messageId] = ev.metadata;
```

Comment:
Why was this required?

Reply:
yeah, it was not actually required. was changing everywhere to read from metadata instead of message. will change back